### ISSUE TITLE
[2.1] Add ability to add a delay between views from the same session

### ISSUE THREAD

**Issue by cyrildewit** [cyrildewit/eloquent-viewable#70]
## Summmay of Feature

The first version (`1.0`) of this package provided a method called `addPageViewThatExpiresAt`. It accepted an expiry date as first argument. This method checked if the model was stored in the session with an expiry date. If not, the view would be stored in the database.

This feature was removed when I created the second version (`2.0`), but since it was useful to some users I decided to bring an improved version of it back.

## Detailed Description

I've got some ideas in my mind, but I'm interested in some feedback!

```php
// 1. (old way)
// @param  DateTime  $expiryDateTime
$post->addViewThatExpiresAt(Carbon::now()->addHours(3)); // 3 hours after now

// 2.
// @param  int  $minutes
$post->addViewWithDelay(3 * 60); // 3 hours after now

// 3. (same as 2 but with two different behaviours)
// @param  DateTime|int  $delay

// 3.1
$post->addViewWithDelay(2 * 60); // 2 hours after now

// 3.2
$post->addViewWithDelay(Carbon::parse('2018-06-10 12:00:00')); // expires at 2018-06-10 12:00:00
```

_Delay is probably not the right word for this behaviour._

---

Fixes #69

**Comment by zymawy** [comment]
Kind of you to return this cool feature. 
I think delay word `$post->addViewWithDelay();` is fine. you may use keep `$post->keepViewUntil();`.
it would be more acquired.

**Comment by cyrildewit** [comment]
@zymawy @SqrFactor I've released a new version `v2.1.0`. Check https://github.com/cyrildewit/eloquent-viewable#saving-views-with-expiry-date.

Thanks for the input from #69!
